[Shared UX] Move No Data Cards to packages

clintandrewhall opened the pull request:
## Summary

This PR:

- moves the `NoDataCard` and `NoDataElasticAgentCard` to their own packages.
- sets up and integrates Storybook and Jest mocks for `NoDataElasticAgentCard`.
- makes relevant changes to dependent components.

![Jun-15-2022 12-28-28](https://user-images.githubusercontent.com/297604/173889064-a9373b3f-5d2b-48fa-9998-8c7996953863.gif)

majagrubic commented on packages/shared-ux/card/no_data/src/no_data_card.tsx:
```diff
@@ -7,37 +7,56 @@
+   * The button will be hidden completely if `isDisabled=true`
+   */
+  button?: string | ReactNode;
+  /** Remapping `onClick` to any element */
```
What does this mean?

clintandrewhall replied:
The comment was there before, so not sure... I'll revise.

majagrubic commented on packages/shared-ux/card/no_data_elastic_agent/src/no_data_elastic_agent_card.tsx:
```diff
@@ -0,0 +1,48 @@
+
+export type Props = Omit<ComponentProps, 'canAccessFleet'>;
+
+export const NoDataElasticAgentCard = ({
```
Please don't kill me for this now 😬 🙏 
The `NoDataCard` component is not actually being used anywhere besides `ElasticAgentCard`  component ([here](
https://github.com/elastic/kibana/pull/134503/files#diff-71d7709c40cda08c59a9bf6539a525a4ca2fb529984762a78ff53196be55bd18L82))

It used to be, but since we've cleaned up some things inside `PageTemplate`, it is not. I've been meaning to clean that up, but didn't consider it high priority. 
How about we remove that component now and put it inside `NoDataElasticAgentCard`? There is no need for it to be a standalone component, and then we could then name *this* component `NoDataCard`. I think name `NoDataElasticAgentCard` exposes too much implementation details anyway.

clintandrewhall replied:
Sure, I can do that.

majagrubic commented on packages/shared-ux/card/no_data/src/no_data_card.tsx:
```diff
@@ -0,0 +1,41 @@
+
+export type Props = Omit<ComponentProps, 'canAccessFleet'>;
+
+export const NoDataCard = ({ href: srcHref, category, description, ...props }: Props) => {
```
🙌

majagrubic reviewed (approved):
Tested this in storybook in Chrome on Mac OSX, works as expected. 
Made a quick pass through the code, but not too detailed, since it's just moving files around I assume there were no major changes.

clintandrewhall commented:
It took more work to get this over the line than I would have liked... in commits since @majagrubic approved this PR, I:

- I adjusted how and where contexts are combined.  I realize these contexts are only going to be in this configuration for a short period, but I still would like a sanity check.
- you can't use `shallow` with `useContext`, so the snapshot for `NoDataPage` was enormous, and I had to remove it. [issue](https://github.com/enzymejs/enzyme/issues/2176)
- I fixed a React props bug with `RedirectAppLinks` that was causing props to be applied (however benignly) to a `div` element.
- I added React `strict` mode to our Storybooks to ensure we don't break React norms.

So I'm requesting @elastic/shared-ux have another look at this before I commit it.  Thanks!

majagrubic commented on packages/shared-ux/link/redirect_app/src/redirect_app_links.stories.tsx:
```diff
@@ -24,16 +25,10 @@ export default {
   return (
     <>
-      <RedirectAppLinks {...{ currentAppId, navigateToUrl }}>
+      <Component {...getStoryServices()}>
```
Nice catch!

majagrubic reviewed (approved):
Code looks good, haven't tested it this time.
